src/main/java/io/debezium/connector/jdbc/SinkRecordDescriptor.java

```diff
@@ -282,8 +290,9 @@ public SinkRecordDescriptor build() {
             Objects.requireNonNull(sinkRecord, "The sink record must be provided.");
 
             final boolean flattened = !isTombstone(sinkRecord) && isFlattened(sinkRecord);
-            readSinkRecordKeyData(sinkRecord, flattened);
-            readSinkRecordNonKeyData(sinkRecord, flattened);
+            final boolean truncated = !flattened && isTruncateEvent(sinkRecord);
+            readSinkRecordKeyData(sinkRecord, flattened, truncated);
```

Comment:
What do you think to the check here about the `truncated` and remove it from the `readSinkRecord*` methods?